facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java

```diff
@@ -0,0 +1,170 @@
+        envs.add(env);
+        fillRepository(envs);
+    }
+
+    @Test
+    public void testGetAll() {
+        Page<Environment> all = provider.getAll(0, 10, null, null);
+
+        assertThat(all.getContent())
+                .hasSize(4)
+                .haveExactly(1, new Condition<>(e -> e.getName().equals(env.getName()), "Environment present"));
+    }
+
+    @Test
+    public void testStore() {
+        final String name = "Hello";
+        final String sysImageId = "NewID";
+
+        Environment environment = Environment.builder().name(name).systemImageId(sysImageId).systemImageType(SystemImageType.DOCKER_IMAGE).build();
+
+        Environment stored = provider.store(environment);
+
+        assertThat(stored.getName()).isEqualTo(name);
+        assertThat(stored.getId()).isNotNull();
+        assertThat(stored.getSystemImageId()).isEqualTo(sysImageId);
+        assertThat(stored.getSystemImageType()).isEqualTo(SystemImageType.DOCKER_IMAGE);
+        Mockito.verify(repository).save(ArgumentMatchers
+                .argThat(env -> name.equals(env.getName())));
+    }
+
+    @Test
+    public void testGetSpecific() {
+        Environment environment = provider.getSpecific(env.getId().toString());
+
+        assertThat(environment.getId()).isEqualTo(env.getId().toString());
+        assertThat(environment.getName()).isEqualTo(env.getName());
+    }
+
+    @Test
+    public void testStoreWithId(){
+        final String name = "Hello";
+        final String sysImageId = "sysID";
+
+        Environment environment = Environment.builder().id("3113").name(name).systemImageId(sysImageId).systemImageType(SystemImageType.DOCKER_IMAGE).build();
+
+        try{
+            provider.store(environment);
+            fail("Validation exception expected");
+        }catch(DTOValidationException ex){
+            // ok
+        }
+    }
+
+    @Test
+    public void testUpdate(){
+        final String newName = "newName";
+
+        Environment environment = Environment.builder()
+                .id(env.getId().toString())
+                .name(newName)
+                .systemImageId(env.getSystemImageId())
+                .deprecated(env.isDeprecated())
+                .systemImageType(env.getSystemImageType())
+                .description(env.getDescription())
+                .systemImageRepositoryUrl(env.getSystemImageRepositoryUrl())
+                .attributes(env.getAttributes())
+                .build();
+
+        assertThat(env.getName()).isNotEqualTo(environment.getName());
+
+        Environment updated = provider.update(env.getId().toString(), environment);
+
+        assertThat(updated.getId()).isEqualTo(env.getId().toString());
+        assertThat(updated.getName()).isNotEqualTo(env.getName());
+        assertThat(updated.getSystemImageType()).isEqualTo(env.getSystemImageType());
+        assertThat(updated.getSystemImageId()).isEqualTo(env.getSystemImageId());
+    }
+
+    @Test
+    public void testDelete(){
```

Comment:
I think we don't allow deleting of environments via REST.

Reply:
YEP